enhancements/machine-api/cluster-api-integration.md

```diff
@@ -0,0 +1,305 @@
+
+Cluster API will only be present in the cluster (installed by a new operator) if and when a user installs a feature gate.
+We will introduce a new, OpenShift specific, feature gate `ClusterAPIEnabled` and include it within the `TechPreviewNoUpgrade` FeatureSet.
+
+Once installed, this preview will allow the user to create new MachineSets using CAPI and explore the
+features available within CAPI, for comparison with MAPI. For example, availability set support in Azure that
+was already in CAPI, but only being introduced to MAPI as of 4.10.
+
+During this timeframe, any user wanting to migrate or try out the preview will be left to manually migrate the MachineSet
+or create a new one, using either the upstream documentation or documentation provided by OpenShift.
+
+This preview will be intended to be used to create day 2 worker MachineSets and is not expected to be integrated into the install process in any way.
+
+It's important to note that the preview will focus on supporting CRDs that our users are familiar with, that includes:
+Machines, MachineSets, MachineHealthChecks. We are not planning to document support for other CRDs like MachineDeployment,
+however they will be installed.
+
+### Supported platforms
+
+The technical preview aims to support: AWS, Azure, GCP, Baremetal, Openstack.
+
+#### Cluster API resource management
+
+In order to maintain the lifecycle of Cluster API related resources, we will create a new operator `cluster-capi-operator`, this name was chosen for avoiding confusion with upstream Cluster API operator.
+This operator will be responsible for all administrative tasks related to the deployment of the Cluster API project within the cluster.
+During tech preview phase, the new operator will also manage all Cluster API related CRDs. All CRD manifests will placed in
+openshift forks of CAPI and will take from there with no changes.
+
+`cluster-capi-operator` and it's operands will be provisioned in a new `openshift-cluster-api` namespace.
+
+The operator will perform the following tasks:
+
+##### Reconcile FeatureGate object
+
+While Cluster API intergration is in tech preview, the operator will reconcile the cluster [`FeatureGate`](https://docs.openshift.com/container-platform/4.8/nodes/clusters/nodes-cluster-enabling-features.html) object and check for `ClusterAPIEnabled` feature gate presence.
+The operator will procceed with Cluster API installation if and only if the required feature gate is present.
+
+##### Deploy Cluster Machine Approver
+
+In order for Cluster API machines to succefully join the cluster, the Kubelet CSRs need to be approved.
+The operator will deploy a separate instance of the cluster-machine-approver, which will be configured to be used with Cluster API machines by providing [`--apigroup`](https://github.com/openshift/cluster-machine-approver/blob/master/main.go#L54) flag that was recently introduced.
```

Comment:
since we use `code blocks` around the capi operator references earlier, i think we should use them here for the cma
```suggestion
The operator will deploy a separate instance of the `cluster-machine-approver`, which will be configured to be used with Cluster API machines by providing [`--apigroup`](https://github.com/openshift/cluster-machine-approver/blob/master/main.go#L54) flag that was recently introduced.
```